### ISSUE TITLE
Improve debug logging for invalid request case

### ISSUE THREAD

**Issue by RobertCraigie** [stoplightio/prism#2265]
## Context

I'm trying to debug invalid request errors & it's easier / faster to look through the server logs than the generated response.

## Current Behavior

Currently if an invalid request is received the server logs something like this with `-v debug`:

```
[13:24:27] ›     [VALIDATOR] ⚠  warning   Request did not pass the validation rules
```

## Expected Behavior

The logs should include any validation error messages as well.

```
[13:24:27] ›     [VALIDATOR] ⚠  error Missing required parameter `foo.bar`
[13:24:27] ›     [VALIDATOR] ⚠  error Expected parameter `foo.baz` to be of type `number` but received `string`
```

**Comment by chohmann** [comment]
@RobertCraigie thank for submitting this! We think this is a great idea and can take advantage of the [recent addition of verbose logging options](https://github.com/stoplightio/prism/pull/2231). We will gladly accept PRs to accomplish this, but will likely not have time to work on this ourselves.

**Comment by daniel-white** [comment]
released in 4.14 #2280 was the implementation
